### PR TITLE
15.0 fix inheritance seller wt

### PR DESCRIPTION
**Pull request by wtaferner**
**Description of the issue/feature this PR addresses:**
Ease the pain on inheriting the method select seller...

**Current behavior before PR:**
It works as expected if you do not need to inherit

**Desired behavior after PR is merged:**
It works as expected even if you need to inherit and can do beautiful things you could not have done before

Info: @wt-io-it

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/product/models/product.py [hunk 1]
```diff
@@ -651,16 +651,17 @@ def open_product_template(self):
...
             date = fields.Date.context_today(self)
         precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
 
-        res = self.env['product.supplierinfo']
-        sellers = self._prepare_sellers(params)
-        sellers = sellers.filtered(lambda s: not s.company_id or s.company_id.id == self.env.company.id)
-        for seller in sellers:
+        sellers_filtered = self._prepare_sellers(params)
+        if self.env.context.get('force_company'):
+            sellers_filtered = sellers_filtered.filtered(lambda s: not s.company_id or s.company_id.id == self.env.context['force_company'])
```
[the remark this answers is not in the document]
Comment (wtaferner) [reply addons/product/models/product.py:662]: Jep, you are right, I think this patch is soo old that I still have the old way in it, I will change that.

File: addons/sale_purchase/models/sale_order.py [hunk 2]
```diff
@@ -277,7 +277,8 @@ def _purchase_service_create(self, quantity=False):
         for line in self:
             line = line.with_company(line.company_id)
             # determine vendor of the order (take the first matching company and product)
-            suppliers = line.product_id._select_seller(quantity=line.product_uom_qty, uom_id=line.product_uom)
+            suppliers = line.product_id.with_company(line.company_id)._select_seller(
+                partner_id=self._context.get('supplier_map', {}).get(line.id), quantity=line.product_uom_qty, uom_id=line.product_uom)
```
Comment (Feyensv) [inline addons/sale_purchase/models/sale_order.py:281]: Unnecessary `with_company` here, the company of `line` environment is already the right one (see line 278)

File: addons/sale_purchase/models/sale_order.py [hunk 3]
```diff
@@ -277,7 +277,8 @@ def _purchase_service_create(self, quantity=False):
         for line in self:
             line = line.with_company(line.company_id)
             # determine vendor of the order (take the first matching company and product)
-            suppliers = line.product_id._select_seller(quantity=line.product_uom_qty, uom_id=line.product_uom)
+            suppliers = line.product_id._select_seller(
+                partner_id=self._context.get('supplier_map', {}).get(line.id), quantity=line.product_uom_qty, uom_id=line.product_uom)
```
Comment (Feyensv) [inline addons/sale_purchase/models/sale_order.py:281]: Where is this context coming from ?

**Review by Feyensv — COMMENTED** [review]
I'm not really knowledgeable (nor responsible) for supplierinfo's logic, I mainly gave the previous remark out of technical curiosity (and `with_company` stuff is my domain) but I'll leave the final review and validation to logistics team ;).

**Comment by Whenrow** [comment]
robodoo r+
Lgtm

**Comment by wtaferner** [comment]
@robodoo r+ rebase-ff

**Comment by wtaferner** [comment]
@Whenrow 
the bot does not like me 😢

**Comment by Whenrow** [comment]
robodoo rebase-ff

**Comment by wtaferner** [comment]
@Whenrow 
Any idea why forward porting is not yet triggered?

**Comment by Whenrow** [comment]
@wtaferner it's temporary deactivated for the 16.4 freeze (it should be done next week) the fw PRs will all be created after that

**Pull request closed without merging**